cpp/oneapi/dal/test/engine/fixtures.hpp

```diff
@@ -89,6 +89,10 @@ class float_algo_fixture : public algo_fixture {
         return is_double && !this->get_policy().has_native_float64();
     }
 
+    bool running_on_gpu() {
```

Comment:
Please move this function into linear_regression fixture: https://github.com/david-cortes-intel/oneDAL/blob/overdetermined_minimum_norm/cpp/oneapi/dal/algo/linear_regression/test/fixture.hpp 

Also, it would be better to rename to something more descriptive like: "non_psd_system_not_supported_on_device"

Reply:
Moved to that file and renamed it.